src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3723,6 +3723,30 @@ size_t os::Linux::find_default_large_page_size() {
   return large_page_size;
 }
 
+void os::Linux::register_large_page_sizes() {
```

Comment:
Please refactor to remove duplicated code with` find_large_page_size`, probably use `register_large_page_sizes` to eliminate the need for `find_large_page_size`

Reply:
In latest patch I removed Linux::find_large_page_size and use register_large_page_sizes. I tried to streamline Linux::setup_large_page_size.

Reply:
with those changes, you have created a bug on os::large_page_size(), I don't think _large_page_size is set (unless I missed it).

Reply:
You are correct. That escaped me, even though I was looking for where os::large_page_size() was set. :)